File: 2025/python/day_1.py
from aoc import *

data = import_input(1)
# data = """L68
# L30
# R48
# L5
# R60
# L55
# L1
# L99
# R14
# L82"""

lines = data.split("\n")
start = 50
zeros = 0

for line in lines:
    d = line[0]
    n = int(line[1:])
    if d == "R":
        start = (start + n) % 100
    elif d == "L":
        start = (start - n) % 100
    if start == 0:
        zeros += 1

print(zeros)


start = 50
zeros = 0
for line in lines:
    d = line[0]
    n = int(line[1:])
    if d == "R":
        for i in range(n):
            start = (start + 1) % 100
            if start == 0:
                zeros += 1
    elif d == "L":
        for i in range(n):
            start = (start - 1) % 100
            if start == 0:
                zeros += 1
    

print(zeros)

File: 2025/c/day_1.c

#include "aoc.h"

int main() {
    char* input = read_input_file("../inputs/day_1.txt");
    StringArray lines = split_string(input, "\n");
    
    int start = 50;
    int zeros = 0;
    for (int i = 0; i < lines.size; i++) {
        char d = lines.array[i][0];
        int val = get_nums(lines.array[i]).array[0];
        if (d == 'R') {
            start = (start + val) % 100;
            if (start == 0) {
                zeros++;
            }
        } else if (d == 'L') {
            start = (start - val) % 100;
            if (start == 0) {
                zeros++;
            }
        }
    }
    printf("\r\n%i\r\n", zeros);


    start = 50;
    zeros = 0;
    for (int i = 0; i < lines.size; i++) {
        char d = lines.array[i][0];
        int val = get_nums(lines.array[i]).array[0];
        if (d == 'R') {
            for (int j = 0; j < val; j++) {
                start++;
                if (start > 99) {
                    start = 0;
                }
                if (start == 0) {
                    zeros++;
                }
            }
        } else if (d == 'L') {
            for (int j = 0; j < val; j++) {
                start--;
                if (start < 0) {
                    start = 99;
                }
                if (start == 0) {
                    zeros++;
                }
            }
        }
    }
    printf("%i\r\n", zeros);

    return 0;
}